src/engine/plugins/match.cpp

```diff
@@ -123,6 +125,16 @@ Status MatchPlugin::HandleRequest(const std::shared_ptr<datafacade::BaseDataFaca
         return Error("InvalidValue", "Invalid coordinate value.", json_result);
     }
 
+    // Check for same or increasing timestamps. Impl. note: Incontrast to `sort(first,
+    // last, less_equal)` checking `greater` in reverse meets irreflexive requirements.
+    const auto time_increases_monotonically = std::is_sorted(
+        parameters.timestamps.rbegin(), parameters.timestamps.rend(), std::greater<>{});
+
+    if (!time_increases_monotonically)
+    {
+        return Error("InvalidValue", "Invalid timestamp value - jumps into the past.", json_result);
```

Comment:
Maybe an error message like "Timestamp values need to increase in time" would work better?